lib/docker/image.rb

```diff
@@ -227,7 +227,16 @@ def search(query = {}, connection = Docker.connection, creds = nil)
     # Import an Image from the output of Docker::Container#export. The first
     # argument may either be a File or URI.
     def import(imp, opts = {}, conn = Docker.connection)
-      open(imp) do |io|
+      require 'open-uri'
+
+      # This differs after Ruby 2.4
+      if URI.public_methods.include?(:open)
```

Comment:
Is this just `#respond_to?`?

Reply:
It's not actually. It `respond_to?` returns true in both pre 2.4 and post 2.4 so we need to check and see if it's an actual public method.